addons/web/static/src/core/expression_editor/expression_editor.js

```diff
@@ -1,25 +1,11 @@
 import { Component, onWillStart, onWillUpdateProps } from "@odoo/owl";
 import { getExpressionDisplayedOperators } from "@web/core/expression_editor/expression_editor_operator_editor";
-import {
-    condition,
-    expressionFromTree,
-    treeFromExpression,
-} from "@web/core/tree_editor/condition_tree";
+import { expressionFromTree, treeFromExpression } from "@web/core/tree_editor/condition_tree";
 import { TreeEditor } from "@web/core/tree_editor/tree_editor";
 import { getOperatorEditorInfo } from "@web/core/tree_editor/tree_editor_operator_editor";
-import { getDefaultValue } from "@web/core/tree_editor/tree_editor_value_editors";
-import { getDefaultPath } from "@web/core/tree_editor/utils";
 import { ModelFieldSelector } from "@web/core/model_field_selector/model_field_selector";
 import { _t } from "@web/core/l10n/translation";
 
-function getDefaultCondition(fieldDefs) {
-    const defaultPath = getDefaultPath(fieldDefs);
-    const fieldDef = fieldDefs[defaultPath];
-    const operator = getExpressionDisplayedOperators(fieldDef)[0];
```

Comment:
we missed something:
getExpressionDisplayedOperators is used instead of getDomainDisplayedOperators
and getDefaultCondition is called on this.filteredFields.
The first problem can be handled by passing the prop getDefaultOperator.
The second on can maybe solved by checking the type of fieldDef in getDefaultPath and only accept m2o fields?

Reply:
this means you'll have to rework core/domain_selector/utils.js in the end I think.

Reply:
or maybe we can juste ignore the param defaultCondition in getPathEditorInfo and compute it like it was done in before in ExpressionEditor (since refuse other res models anyway)